### PR TITLE
Source connector doc updates

### LINKED ISSUE #1612: Document the best practice for giving access to 3rd party for kinesis
### Related code PR

_No response_

### Which part(s) of the docs might be affected or should be updated? And how?

Please see the bottom section on this notion page:
https://www.notion.so/risingwave-labs/Access-Kinesis-with-arn-session_token-fbe6d52f33ed495a846b19522e245700

We need to update the Kinesis page - add a paragraph about this best practice.

### Reference

_No response_

### LINKED ISSUE #1759: Bug: incorrect descriptions in Source topics
### Where's the bug?

https://docs.risingwave.com/docs/current/data-ingestion/

### Describe the bug

Please refer to [this page](https://www.notion.so/risingwave-labs/Source-Doc-Suggested-Changes-f36180843ee148feb0b7da0f11b627c6?pvs=4) for details.

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  - Major updates to source connector docs based on https://www.notion.so/risingwave-labs/Source-Doc-Suggested-Changes-f36180843ee148feb0b7da0f11b627c6?pvs=4

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1759, https://github.com/risingwavelabs/risingwave-docs/issues/1612

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/ingest/ingest-from-kafka.md [hunk 1]
```diff
@@ -50,21 +50,6 @@ export const svg = rr.Diagram(
          rr.NonTerminal('source_name', 'skip'),
```
Comment (hengm3467) [inline docs/ingest/ingest-from-kafka.md]: @emile-00 We can remove the railway syntax completely. It's hard to maintain and doesn't provide additional value at the moment. We'll explore a new way (automatically updated) to do the syntax.

File: docs/ingest/ingest-from-kinesis.md [hunk 2]
```diff
@@ -78,6 +63,21 @@ export const svg = rr.Diagram(
                 rr.Terminal(')'),
             ),
         ),
+                rr.Sequence(
```
Comment (hengm3467) [inline docs/ingest/ingest-from-kinesis.md]: Same here. We can remove the railway syntax.

File: docs/ingest/ingest-from-pulsar.md [hunk 3]
```diff
@@ -81,6 +66,21 @@ export const svg = rr.Diagram(
                 ),
                 rr.Terminal(')'),
             ),
+        ),
```
Comment (hengm3467) [inline docs/ingest/ingest-from-pulsar.md]: And here.

**Review by ShanlanLi — APPROVED** [review]
LGTM

File: docs/ingest/ingest-from-gcs.md [hunk 4]
```diff
@@ -7,6 +7,8 @@ slug: /ingest-from-gcs
 
 Use the SQL statement below to connect RisingWave to a Google Cloud Storage source.
 
+The GCS connector does not guarantee the sequential reading of files or complete file reading.
```
Comment (neverchanje) [inline docs/ingest/ingest-from-gcs.md]:
```suggestion
## Load Order of GCS files
The GCS connector does not guarantee the sequential reading of files or complete file reading. 
```

@tabVersion could you elaborate on the ordering and completeness guarantee of GCS source? I think 'complete file reading' is a must-have.

**Review by neverchanje — COMMENTED** [review]
Rest LGTM

File: docs/ingest/ingest-from-gcs.md (hunk 4 above)
Comment (tabVersion) [reply docs/ingest/ingest-from-gcs.md]: > elaborate on the ordering and completeness guarantee of GCS source

We guarantee completeness on each file in the end but we don't always read each file continuously. 
For example, at T0, rw reads file F1 to offset O1 and crashes. Then rw rebuilds the task queue and does not guarantee the next task is file F1.

Comment (hengm3467) [reply docs/ingest/ingest-from-gcs.md]: -> "Loading order of GCS files"
Let's use sentence style, and add the additional information about data ordering and completeness. @emile-00

**Review by hengm3467 — APPROVED** [review]
Otherwise LGTM

**Pull request merged**